### ISSUE TITLE
Files opened as C don't work with SublimeDiscordRP

### ISSUE THREAD

**Issue by Slattz** [Snazzah/SublimeDiscordRP#18]
Hi, I'm experiencing a problem.
Any files that I have marked in Sublime as the `C` Language don't show up in Discord, such as `.c` & `.h`. They don't appear on Discord unless I specify their language as C++ in the `Open all with current extension as...` drop-down but that's not very practical.
Thanks for this plugin btw, apart from this issue, it's awesome 🙂

**Comment by FichteFoll** [comment]
I confirmed that the detection works properly. My guess is that the `lang-c` icon doesn't exist, although it was contained within the icon map before, and Discord just ignores the update with that icon entirely.

@Snazzah could you please take a look at the registered icons? Or give me the required credentials for the discord icon upload so I can check myself?

**Comment by Snazzah** [comment]
@FichteFoll
![screen shot 2018-01-23 at 3 10 32 pm](https://user-images.githubusercontent.com/7025343/35300910-a46a29fc-004f-11e8-85dd-9fd73ae35848.png)

**Comment by Snazzah** [comment]
also @Slattz you can do `Ctrl-Shift-I` to check Discord console, start Sublime, then paste what console says in the issue.

**Comment by Slattz** [comment]
@Snazzah Here's everything in the console after clearing it then opening Sublime:
```

2bb35e4….js:17 [RunningGameStore] games []
2bb35e4….js:17 [RPCServer:IPC] Socket Opened: 3
2bb35e4….js:17 [RPCServer:IPC] Socket Emit: 3 Object {cmd: "DISPATCH", data: Object, evt: "READY", nonce: null}
2bb35e4….js:17 [RPCServer:IPC] Socket Message: 3 Object {args: Object, cmd: "SET_ACTIVITY", nonce: "10f47560-54c0-4825-96b8-7eca743dfeb2"}
2bb35e4….js:17 [RPCServer:IPC] Socket Emit: 3 Object {cmd: "SET_ACTIVITY", data: Object, evt: null, nonce: "10f47560-54c0-4825-96b8-7eca743dfeb2"}cmd: "SET_ACTIVITY"data: Objectevt: nullnonce: "10f47560-54c0-4825-96b8-7eca743dfeb2"__proto__: Object
2bb35e4….js:17 [RunningGameStore] games [Object]
2bb35e4….js:17 [RPCServer:IPC] Socket Message: 3 Object {args: Object, cmd: "SET_ACTIVITY", nonce: "a6894a18-3a17-49f3-b318-4bc325dc3018"}
2bb35e4….js:17 [RPCServer:IPC] Socket Emit: 3 Object {cmd: "SET_ACTIVITY", data: Object, evt: "ERROR", nonce: "a6894a18-3a17-49f3-b318-4bc325dc3018"}
```

**Comment by Snazzah** [comment]
@slattz can you open the data object in the last line>

**Comment by Slattz** [comment]
@Snazzah I couldn't copy and paste it so I took a pic, here you go.
![a](https://user-images.githubusercontent.com/15308186/35302700-5678356c-0087-11e8-8ccc-ea78ad394a99.PNG)

**Comment by Snazzah** [comment]
Fixed in [`cd65795`](https://github.com/Snazzah/SublimeDiscordRP/commit/cd6579588f086f29d6437844407188a8e2cb06e7).

**Comment by FichteFoll** [comment]
Good to know that discord renders these errors in its console. I would have set up some reading for the IPC, but since all I/O is blocking it would probably need a separate thread and then I just didn't consider it worth it anymore. I also didn't know whether a reply was guaranteed.
